src/univariate/continuous/arcsine.jl

```diff
@@ -23,18 +23,20 @@ External links
 
 * [Arcsine distribution on Wikipedia](http://en.wikipedia.org/wiki/Arcsine_distribution)
 
+Use `Arcsine(a, b, ::NoArgCheck)` to bypass argument checks.
 """
 struct Arcsine{T<:Real} <: ContinuousUnivariateDistribution
     a::T
     b::T
-
-    Arcsine{T}(a::T, b::T) where {T} = (@check_args(Arcsine, a < b); new{T}(a, b))
+    Arcsine{T}(a::T, b::T) where {T<:Real} = new{T}(a, b)
 end
 
-Arcsine(a::T, b::T) where {T<:Real} = Arcsine{T}(a, b)
+Arcsine(a::T, b::T) where {T <: Real} = (@check_args(Arcsine, a < b); Arcsine{T}(a, b))
+Arcsine(a::T, b::T, ::NoArgCheck) where {T} = Arcsine{T}(a, b)
+
 Arcsine(a::Real, b::Real) = Arcsine(promote(a, b)...)
 Arcsine(a::Integer, b::Integer) = Arcsine(Float64(a), Float64(b))
-Arcsine(b::Real) = Arcsine(0.0, b)
+Arcsine(b::T) where {T <: Real} = Arcsine(zero(T), b)
```

Comment:
I am not sure this is an OK definition, as later we e.g. define: `kurtosis(d::Arcsine{T}) where {T<:Real} = -T(3/2)`, which will fail if we write `Arcsine(1)`.
A general question is if for continuous distributions, in cases like this one, we should not add restriction for parameters to be `AbstractFloat`?

Reply:
or add a conversion if parameters are `<:Integer` to float?

Reply:
`Arcsine(1)` will dispatch to `Arcsine(a::Integer, b::Integer)`:
```julia
julia> kurtosis(Arcsine(1))
-1.5
```

Seems fine?

Reply:
> or add a conversion if parameters are <:Integer to float?

there is one on line 38

Reply:
OK, I have not noticed it, sorry. It should probably use `float` instead of `Float64` as noted below.